File: September/2026-09-05/65/65-valid-phone-number.py
# 65-valid-phone-number.cpp;string;help;letcode

class Solution:
    def isNumber(self, s: str) -> bool:
        seen_digit = False
        seen_dot = False
        seen_exponent = False
        digit_after_exponent = True

        for i, ch in enumerate(s):
            if ch.isdigit():
                seen_digit = True
                if seen_exponent:
                    digit_after_exponent = True
            elif ch == '.':
                if seen_dot or seen_exponent:
                    return False
                seen_dot = True
            elif ch == 'e' or ch == 'E':
                if seen_exponent or not seen_digit:
                    return False
                seen_exponent = True
                digit_after_exponent = False
            elif ch == '+' or ch == '-':
                if i != 0 and s[i - 1] != 'e' and s[i - 1] != 'E':
                    return False
            else:
                return False
        return seen_digit and digit_after_exponent

File: September/2026-09-05/65/65-valid-phone-number.cpp
65-valid-phone-number.cpp;string;help;letcode

class Solution {
public:
    bool isNumber(string s) {
        bool seenDigit = false;
        bool seenDot = false;
        bool seenExp = false;
        bool digitAfterExp = true;

        for (int i = 0; i < s.size(); i++) {
            char ch = s[i];

            if (isdigit(ch)) {
                seenDigit = true;

                if (seenExp)
                    digitAfterExp = true;
            }
            else if (ch == '+' || ch == '-') {
                if (i > 0 && s[i - 1] != 'e' && s[i - 1] != 'E')
                    return false;
            }

            else if (ch == '.') {
                if (seenDot || seenExp)
                    return false;

                seenDot = true;
            }

            else if (ch == 'e' || ch == 'E') {
                if (seenExp || !seenDigit)
                    return false;

                seenExp = true;
                digitAfterExp = false;
            }

            else {
                return false;
            }
        }

        return seenDigit && digitAfterExp;
    }
};